sdk/python/kfp/compiler/_default_transformers.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ..dsl._container_op import BaseOp
+
+def add_pod_env(op: BaseOp) -> BaseOp:
+    """Adds pod environment info to the operator.
+    """
+    if op.pod_labels and op.pod_labels['add-pod-env'] == 'true':
```

Comment:
I'd suggest adding `isinstance(op, ContainerOp)` to the `if` statement, just to make sure this does not break in case of a user's mistake setting that pod label to every op, and `ResourceOp`s exist in their pipeline.

Reply:
Echo @elikatsis

Reply:
Done